core/src/main/java/io/kestra/core/plugins/PluginScanner.java

```diff
@@ -32,14 +33,37 @@ public PluginScanner(ClassLoader parent) {
         this.parent = parent;
     }
 
+    /**
+     * Scans the specified top-level plugin directory for plugins. Keeps the scan alive allowing plugins hot-reload
+     * @param onNewPlugin the callback to be called when a new plugin is found
+     */
+    public void continuousScan(final Path pluginPaths, Consumer<RegisteredPlugin> onNewPlugin) {
+        try {
+            new PluginResolver(pluginPaths).watch(
```

Comment:
Why is it the plugin resolver that watches and not the scanner.
The scanner, as it names indicate, is responsible or scanning plugins so it may be the best place to watch the plugin dir.

Reply:
I agree but I didn't want to go too big on this and saw that PluginResolver was the entrypoint of actual path / file scanning so I didn't want to move out too much things on this one, maybe a later rework ?